controller/update.go

```diff
@@ -209,6 +232,30 @@ type UpdateOptions struct {
 	ReadySecs int
 }
 
+// updateCurrentSliceIDs updates the list of current slice IDs,
+// removing the slice that was modified, and adding any new slice IDs.
+func (c controller) updateCurrentSliceIDs(ctx context.Context, currentSliceIDs []string, modifiedSliceIDs []string, newSliceIDs []string) []string {
+	c.Config.Logger.Debug(ctx, "current slice IDs: %v", currentSliceIDs)
+	c.Config.Logger.Debug(ctx, "modified slice IDs: %v", modifiedSliceIDs)
+	c.Config.Logger.Debug(ctx, "new slice IDs: %v", newSliceIDs)
+
+	for i, currentSliceID := range currentSliceIDs {
+		for _, modifiedSliceID := range modifiedSliceIDs {
+			if currentSliceID == modifiedSliceID {
+				currentSliceIDs = append(
+					currentSliceIDs[:i],
+					currentSliceIDs[i+1:]...,
+				)
+			}
+		}
+	}
+	currentSliceIDs = append(currentSliceIDs, newSliceIDs...)
+
+	c.Config.Logger.Debug(ctx, "updated slice IDs: %v", currentSliceIDs)
+
+	return currentSliceIDs
+}
+
 func (c controller) UpdateWithStrategy(ctx context.Context, req Request, opts UpdateOptions) error {
```

Comment:
Yeah, it would be cool to improve this, but I'd rather it happen in a separate PR.